WordPress/Classes/ViewRelated/Post/PostEditor+Publish.swift

```diff
@@ -134,20 +131,14 @@ extension PublishingEditor {
         switch action {
         case .schedule, .publish:
             showPrepublishingSheet(for: action, analyticsStat: analyticsStat)
-        case .saveAsDraft:
-            performSaveDraftAction()
         case .update:
-            if post.original().status == .draft {
-                performSaveDraftAction()
-            } else {
-                performUpdatePostAction()
-            }
+            performUpdatePostAction()
         case .submitForReview:
             // TODO: Show Prepublishing VC
             fatalError("Not implemented (kahu-offline-mode)")
-        case .save, .continueFromHomepageEditing:
-            assertionFailure("No longer used and supported")
+        case .save, .saveAsDraft, .continueFromHomepageEditing:
```

Comment:
If I open a draft as a contributor, the app crashes at this assertion failure due to the post editor action being `.save`

Reply:
I think there are a few issues with the contributor role, which I think is related to the sync engine changes:
* If I create a new draft as a contributor, the post list cell activity indicator never disappears. When I re-open the editor, then tap the back button without making any changes, the app crashes because it fails the `isOriginal()` assertion for `isNewDraft`
* As mentioned above, if I open a draft that's been submitted for review, the app crashes because the post editor action is `.save` (doesn't matter if you're logged in as a contributor or an admin)

Reply:
I'll cover the contributor role and pending posts in a separate PR and according to the amend to the spec 👍 
Working on it right now.

Reply:
Fixed in the following PR: https://github.com/wordpress-mobile/WordPress-iOS/pull/22949